### DISCUSSION TITLE
ignoring vim.o.winwidth

### DISCUSSION THREAD

**Discussion by pigoz** [nvim-tree/nvim-tree.lua#1604] [Q&A]
Hi! I have a smaller screen and `vim.o.winwidth = 80` in my config. That way I can see 80 columns when I change the focused window on vertical splits.

Unfortunately nvim-tree also becomes 80 columns (opened through `:NvimTreeToggle<cr>`), which I don't need. Is there someway to work around this?

**Comment by alex-courtis** [comment]
> Hi! I have a smaller screen and vim.o.winwidth = 80 in my config. That way I can see 80 columns when I change the focused window on vertical splits.

Today I learned!

> Unfortunately nvim-tree also becomes 80 columns

`winwidth` is global so unfortunately we cannot set it just for the nvim-tree window.

Experimentation revealed that we can turn `winwidth` off when the tree is open:
```lua
local events = require('nvim-tree.api').events
local Event = events.Event
events.subscribe(Event.TreeOpen, function()
  vim.o.winwidth = 20
end)
events.subscribe(Event.TreeClose, function()
  vim.o.winwidth = 80
end)
```

**Comment by pigoz** [comment]
Thank you! This looks very promising. But instead of the tree being open it should probably listen on the tree being focused.

For the time being I solved my issue with this plugin: https://github.com/beauwilliams/focus.nvim

**Reply by alex-courtis** [reply]
Interesting `:FocusSplitNicely`
